app/src/main/java/de/danoeh/antennapod/ui/screen/playback/TranscriptFragment.java

```diff
@@ -0,0 +1,290 @@
+import android.graphics.drawable.ColorDrawable;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import androidx.fragment.app.DialogFragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.LinearSmoothScroller;
+import androidx.recyclerview.widget.RecyclerView;
+
+import android.text.TextUtils;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.WindowManager;
+import android.widget.ProgressBar;
+import android.widget.Toast;
+
+import com.google.android.material.dialog.MaterialAlertDialogBuilder;
+
+import de.danoeh.antennapod.databinding.TranscriptDialogBinding;
+import de.danoeh.antennapod.playback.base.PlayerStatus;
+import de.danoeh.antennapod.playback.service.PlaybackController;
+import de.danoeh.antennapod.ui.chapters.PodcastIndexTranscriptUtils;
+import org.greenrobot.eventbus.EventBus;
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.event.playback.PlaybackPositionEvent;
+import de.danoeh.antennapod.model.feed.FeedMedia;
+import de.danoeh.antennapod.model.feed.Transcript;
+import de.danoeh.antennapod.model.feed.TranscriptSegment;
+import de.danoeh.antennapod.model.playback.Playable;
+import de.danoeh.antennapod.ui.common.ThemeUtils;
+import de.danoeh.antennapod.ui.transcript.TranscriptViewholder;
+import io.reactivex.Maybe;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+
+public class TranscriptFragment extends DialogFragment {
+    public static final String TAG = "TranscriptFragment";
+    private TranscriptDialogBinding viewBinding;
+    private PlaybackController controller;
+    private Disposable disposable;
+    Playable media;
+    Transcript transcript;
+    TreeMap<Long, TranscriptSegment> segmentsMap;
+    private ProgressBar progressBar;
+    TranscriptAdapter adapter = null;
+    TranscriptViewholder currentView = null;
+    TranscriptViewholder prevView = null;
+    RecyclerView viewTranscript;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    public TranscriptFragment initMedia(Playable m) {
+        media = m;
+        return this;
+    }
+
+    @Override
+    public void onResume() {
+        ViewGroup.LayoutParams params;
+        params = getDialog().getWindow().getAttributes();
+        params.width = WindowManager.LayoutParams.MATCH_PARENT;
+        getDialog().getWindow().setAttributes((WindowManager.LayoutParams) params);
+        getDialog().getWindow().setBackgroundDrawable(new ColorDrawable(Color.BLACK));
```

Comment:
@ByteHamster - any tip on how to get the background color for the theme?

this didn't work

```
        getDialog().getWindow().setBackgroundDrawableResource(ThemeUtils.getDrawableFromAttr(getActivity(),
                R.attr.selectableItemBackground)
        );```

Reply:
Hmm, it feels a bit strange that this needs to be set manually. Can't you just remove setting the window background? Then the Material3 library handles the color.

Reply:
removed the need for custom setting, it was because of the `AppCompatDialogFragment`subclass (which I removed)